solidity/test/TokenBridgeRouter.t.sol

```diff
@@ -0,0 +1,206 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import {TokenBridgeRouter} from "../contracts/middleware/token-bridge/TokenBridgeRouter.sol";
+import {MockToken} from "../contracts/mock/MockToken.sol";
+import {TestTokenRecipient} from "../contracts/test/TestTokenRecipient.sol";
+import {MockTokenBridgeAdapter} from "../contracts/mock/MockTokenBridgeAdapter.sol";
+import {HyperlaneTestHelper} from "./HyperlaneTestHelper.t.sol";
+
+import {TypeCasts} from "../contracts/libs/TypeCasts.sol";
+
+contract TokenBridgeRouterTest is Test, HyperlaneTestHelper {
+    TokenBridgeRouter originTokenBridgeRouter;
+    TokenBridgeRouter destinationTokenBridgeRouter;
+
+    // Origin bridge adapter
+    MockTokenBridgeAdapter bridgeAdapter;
+    string bridge = "FooBridge";
+
+    uint32 originDomain = 123;
+    uint32 destinationDomain = 321;
+
+    TestTokenRecipient recipient;
+    MockToken token;
+    bytes messageBody = hex"beefdead";
+    uint256 amount = 420000;
+
+    event TokenBridgeAdapterSet(string indexed bridge, address adapter);
+
+    function setUp() public {
+        token = new MockToken();
+        bridgeAdapter = new MockTokenBridgeAdapter(token);
+        recipient = new TestTokenRecipient();
+
+        originTokenBridgeRouter = new TokenBridgeRouter();
+        destinationTokenBridgeRouter = new TokenBridgeRouter();
+
+        hyperlaneTestHelperSetUp(originDomain, destinationDomain);
+
+        // TODO: set IGP?
+        originTokenBridgeRouter.initialize(
+            address(this),
+            address(originManager),
```

Comment:
Favoring inheritance over composition makes this difficult to read. Consider instead composing HyperlaneTestEnvironment and having view calls on that (e.g. hyperlaneTestEnvironment.abacusConnectionManager(origin))